dev/PushNotifications/PushNotificationsLongRunningTask.StartupTask/winmain.cpp

```diff
@@ -0,0 +1,47 @@
+﻿#include <windows.h>
+#include "pch.h"
+
+#include "../PushNotifications-Constants.h"
+
+#include <NotificationsLongRunningProcess_h.h>
+#include <wil/resource.h>
+
+inline bool isRetriableRpcError(HRESULT hr)
+{
+    return (hr == HRESULT_FROM_WIN32(EPT_S_NOT_REGISTERED)) ||
+        (hr == HRESULT_FROM_WIN32(RPC_S_SERVER_UNAVAILABLE)) ||
+        (hr == HRESULT_FROM_WIN32(RPC_S_CALL_FAILED)) ||
+        (hr == HRESULT_FROM_WIN32(RPC_S_CALL_FAILED_DNE)) ||
+        (hr == HRESULT_FROM_WIN32(RPC_S_UNKNOWN_IF));
+}
+
+int WINAPI WinMain(HINSTANCE /*hInstance*/, HINSTANCE /*hPrevInstance*/, PSTR /*lpCmdLine*/, int /*nCmdShow*/)
+{
+    RETURN_IF_FAILED(::CoInitializeEx(nullptr, COINITBASE_MULTITHREADED));
+
+    auto scopeExit = wil::scope_exit(
+        [&]() { CoUninitialize(); });
+
+    unsigned int retries = 0;
+    bool isRpcRetriableError = true;
+    HRESULT hr = S_OK;
+
+    while (retries < 3 && isRpcRetriableError)
+    {
+        try
+        {
+            wil::com_ptr<INotificationsLongRunningPlatform> longRunningProcessPlatform
+                { wil::CoCreateInstance<NotificationsLongRunningPlatform, INotificationsLongRunningPlatform>(CLSCTX_LOCAL_SERVER) };
+
+            isRpcRetriableError = false;
+        }
+        catch (...)
+        {
+            hr = wil::ResultFromCaughtException();
+            isRpcRetriableError = isRetriableRpcError(hr);
```

Comment:
This addresses only start-up failures but does not address a crash and restart.  In a subsequent PR you need to create a scheme to monitor process lifetime and restart.  Unless you can depend on the next registration to restart the LRP.  But this will probably lose notifications so we probably need a monitoring and restart system.

Reply:
@EHO-makai This is a great point! Right now, if the LRP goes down, only the next app registration will wake it up again. I like the idea of enhancing the startup task to poll the LRP status and launch it again in case it's not there.

I have filed a deliverable for it -> #1360